projects/VitalRecord/VitalRecord_util.cs

```diff
@@ -94,6 +94,143 @@ protected void UpdateEntry(bool shouldExist, [CallerMemberName] string propertyN
+            if (entry != null)
+            {
+                Observation observation = (Observation)entry.Resource;
+                if (extensionURL != null)
+                {
+                    Extension extension = observation?.Value?.Extension.FirstOrDefault(ext => ext.Url == extensionURL);
+                    if (extension != null && extension.Value != null && extension.Value.GetType() == typeof(CodeableConcept))
+                    {
+                        return CodeableConceptToDict((CodeableConcept)extension.Value);
+                    }
+                }
+                else
+                {
+                    return CodeableConceptToDict((CodeableConcept)observation.Value);
+                }
+            }
+            return EmptyCodeableDict();
+        }
+
+        /// <summary>Helper to support vital record property setter helper methods for values stored in Observations.</summary>
+        /// <param name="value">the coded value to set as the value of the property</param>
+        /// <param name="code">the code to specify the type of Observation</param>
+        /// <param name="codeSystem">the code system of the code specifying the type of Observation</param>
+        /// <param name="text">the text for the code specifying the type of Observation</param>
+        /// <param name="profileURL">the profile URL to include in the meta of the Observation</param>
+        /// <param name="section">the section of the composition the Observation should be added to</param>
+        /// <param name="extensionURL">if present, specifies that the value should be set on an extension with the provided URL instead</param>
+        /// <param name="propertyName">the name of the C# property, used to determine the subject ID</param>
+        protected void SetObservationValue(Dictionary<string, string> value, string code, string codeSystem, string text, string profileURL, string section, string extensionURL = null, [CallerMemberName] string propertyName = null)
+        {
+            var entry = Bundle.Entry.Where(e => e.Resource is Observation obs && CodeableConceptToDict(obs.Code)["code"] == code).FirstOrDefault();
+            Observation observation;
+            // If the observation is there we use it, otherwise create it
+            if (entry != null)
+            {
+                observation = (Observation)entry.Resource;
+            }
+            else
+            {
+                observation = new Observation();
+                observation.Id = Guid.NewGuid().ToString();
+                observation.Meta = new Meta();
+                string[] profile = { profileURL };
+                observation.Meta.Profile = profile;
+                observation.Code = new CodeableConcept(codeSystem, code, text, null);
+                observation.Subject = new ResourceReference($"urn:uuid:{SubjectId(propertyName)}");
```

Comment:
Some observations set the Subject and a few set the Focus, we may need to add observation.Focus to this as well.

Reply:
Yes, definitely, lets add these as needed